### PR TITLE
Fix vulnerability in Python's cryptography library

### LINKED ISSUE #6442: [FIX] Vulnerability in Python's cryptography library
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.0.0 | API/Framework | Manager | Packages/Sources | Linux |

Hi team,

Recently a vulnerability has been detected in thePython's cryptography library. This is described as follows:

`Affected versions of this package are vulnerable to Timing Attack. It is vulnerable to Bleichenbacher timing attacks in the RSA decryption API, via timed processing of valid PKCS#1 v1.5 ciphertext.`

This is fixed in cryptography 3.2. pyca/cryptography@58494b4 is the resolving commit. So we are going to upload the required version to 3.2 or higher, this will be based on our internal testing.

Regards

### PR DESCRIPTION
**Pull request by AdriiiPRodri**
|Related issue|
|---|
|#6442|

Hi team,

This PR closes #6442. In this PR we have updated the cryptography library version because a vulnerability has been found in the version we were using.

```
adriiiprodri@wazuh python3 run_tests.py -rbac both -k agents    
Collected tests [6]:
test_agents_DELETE_endpoints.tavern.yaml, test_agents_GET_endpoints.tavern.yaml, test_agents_POST_endpoints.tavern.yaml, test_agents_PUT_endpoints.tavern.yaml, test_rbac_black_agents_endpoints.tavern.yaml, test_rbac_white_agents_endpoints.tavern.yaml


test_agents_DELETE_endpoints.tavern.yaml 
	 7 passed, 8 warnings

test_agents_GET_endpoints.tavern.yaml 
	 90 passed, 91 warnings

test_agents_POST_endpoints.tavern.yaml 
	 5 passed, 6 warnings

test_agents_PUT_endpoints.tavern.yaml 
	 9 passed, 10 warnings

test_rbac_black_agents_endpoints.tavern.yaml 
	 39 passed, 40 warnings

test_rbac_white_agents_endpoints.tavern.yaml 
	 39 passed, 40 warnings
```

Regards

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,20 @@
 # Change Log
 All notable changes to this project will be documented in this file.
 
+## [v4.0.1] -
+
+### Added
+
+### Changed
```
Comment (davidjiglesias) [inline CHANGELOG.md:8]:
```suggestion
### Changed

- **Framework:**
  - Update Python's cryptography library to version 3.2.1 ([#6442](https://github.com/wazuh/wazuh/issues/6442))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
+
+### Fixed
+
+- **Framework:**
```
Comment (davidjiglesias) [inline CHANGELOG.md:12]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
+  - Fix Python's cryptography library ([#6442](https://github.com/wazuh/wazuh/issues/6442))
```
Comment (davidjiglesias) [inline CHANGELOG.md:13]:
```suggestion
```

**Review by davidjiglesias — CHANGES_REQUESTED** [review]
Please run cluster unit tests and integration tests

**Comment by AdriiiPRodri** [comment]
> Please run cluster unit tests and integration tests

## Cluster integration tests
```adriiiprodri@wazuh python3 run_tests.py -rbac both -k cluster                  
Collected tests [3]:
test_cluster_endpoints.tavern.yaml, test_rbac_black_cluster_endpoints.tavern.yaml, test_rbac_white_cluster_endpoints.tavern.yaml


test_cluster_endpoints.tavern.yaml 
	 56 passed, 57 warnings

test_rbac_black_cluster_endpoints.tavern.yaml 
	 21 passed, 22 warnings

test_rbac_white_cluster_endpoints.tavern.yaml 
	 21 passed, 22 warnings```

**Review by davidjiglesias — APPROVED** [review]

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**